test/suite/test_cursor_details01.py

```diff
@@ -0,0 +1,34 @@
+
+import time, wttest
+
+class test_cursor_details(wttest.WiredTigerTestCase):
+    """
+    Check that the cursor get_details API works with a checkpoint cursor.
+    """
+
+    def test_cursor_details(self):
+
+        # Create a checkpoint.
+        uri = 'table:cursor_details_checkpoint'
+        internal_checkpoint_name = 'WiredTigerCheckpoint'
+        self.session.create(uri, "key_format=S,value_format=S")
+        self.session.checkpoint()
+
+        # Query checkpoint cursor details. This is the first and currently
+        # only cursor to support this API all other should return ENOTSUP.
+        cursor = self.session.open_cursor(uri, None, f"checkpoint={internal_checkpoint_name}")
+        ret, details = cursor.get_details(None)
+        
```

Comment:
It would be nice to check that we're doing the two things we're expected to do.  That is:

- add some data, advance stable timestamp, do a checkpoint, open a cursor on that checkpoint and validate the timestamp.
- do a checkpoint, open a cursor on most recent checkpoint, open a second cursor on recent checkpoint, validate they have the same id, do another checkpoint and open a third cursor, and validate they have diff ids.